src/elm/clinical.ts

```diff
@@ -2,15 +2,17 @@ import { Expression } from './expression';
 import * as dt from '../datatypes/datatypes';
 import { Context } from '../runtime/context';
 import { build } from './builder';
+import ELM from '../types/elm';
 
 export class ValueSetDef extends Expression {
   name: string;
   id: string;
   version?: string;
 
-  constructor(json: any) {
+  constructor(json: ELM.ValueSetDef) {
+    // @ts-ignore
```

Comment:
Looks like this `ts-ignore` is because of issues with the ELM types? Is this a known issue? If so, we should maybe add a comment explaining the presence of this, and what the discrepancy of the ELM types are. To make sure `eslint` doesn't fail, you should also disable the lint rule above this line:

```suggestion
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
```

Reply:
So this is a tricky one. @cmoesel also including you here. The issue is that the `Def` classes ValueSetDef, ConceptDef, CodeSystemDef, etc. DO NOT inherit from `Expression` in the XML schema data. Yet the class in the TypeScript does.
```
<xs:complexType name="ValueSetDef">
<xs:complexContent>
	<xs:extension base="Element">
		<xs:sequence>
			<xs:element name="codeSystem" type="CodeSystemRef" minOccurs="0" maxOccurs="unbounded">
			</xs:element>
		</xs:sequence>
		<xs:attribute name="name" type="xs:string"/>
		<xs:attribute name="id" type="xs:string" use="required">
		</xs:attribute>
		<xs:attribute name="version" type="xs:string" use="optional">
		</xs:attribute>
		<xs:attribute name="accessLevel" type="AccessModifier" use="optional" default="Public"/>
	</xs:extension>
</xs:complexContent>
</xs:complexType>

export class ValueSetDef extends Expression {
  name: string;
  id: string;
  version?: string;

  constructor(json: ELM.ValueSetDef) {
    // @ts-ignore
    super(json);
    this.name = json.name || '';
    this.id = json.id;
    this.version = json.version;
  }
etc...
}

```
Notice the `base='Element'`. In most other "expressions", the XML looks has `base='Expression'`: 
```

<xs:complexType name="MinValue">
<xs:complexContent>
	<xs:extension base="Expression">
		<xs:attribute name="valueType" type="xs:QName" use="required"/>
	</xs:extension>
</xs:complexContent>
</xs:complexType>


```
To further complicate, the generated ELM for the `Def` expressions do not carry a `type` property that allows the differentiation of an expression type. 
```
// No Type
interface ValueSetDef extends AbstractElement {
      name?: string;
      id: string;
      version?: string;
      accessLevel?: AccessModifier;
      codeSystem?: CodeSystemRef[];
}

// Has Type
interface MinValue extends AbstractExpression {
      type: "MinValue";
      valueType: string;
}
```
Which is necessary to differentiate the different types of Expressions currently being evaluated using 
```   
type Expression =
      | OperatorExpression
      | ExpressionRef
      | FunctionRef
      | ParameterRef
      | + a bunch more
```
So to summarize, the problem could possibly be solved by simply adding 'base=Expression' to the def in the XML, but 

1. Not sure the Core Java Code wants us to keep changing the XML spec for this library (maybe through an MR)? There would then likely be changes to the generated ELM (adding of the type key for expression differentiation) which could potentially break things. Also that may not make logical sense anyways, because a `ValueSetDef` can not exist anywhere in the Expression tree, only at the root ELM.Library layer.
2. Could change this code a bit on the JS side, ensuring that these `Def` classes do not inherit from Expression. But of course we would need to think through that a bit more than just adding `// @ts-ignore` as I did here.

Reply:
Hey @cmoesel, interfaces would definitely be an interesting approach to explore. It makes sense that switching to TypeScript might lean the library towards certain design patterns that might have been handled a little differently in vanilla JavaScript. 

Sorry I haven't made any more progress on this MR, I've actually been working on another related project that you may find interesting though! It's a browser based CQL IDE and publisher tool that provides support for real-time validation of CQL, CQL statement type checking, dependency validation and resolution, and other tooling that supports writing CQL. It uses cql-exec under the hood for CQL execution. 

If you're interested, I'd love to share the progress and maybe get some feedback from you and @mgramigna in a week or two and see what you guys think about the approach?

P.S. - It also may be worth looking into the type support for the cql execution results. One thing I'm seeing with fhir-exec is the results are not in FHIR format. For instance, a codeable_concept.code has a `.value` property, which in FHIR is just a string.

Reply:
Hi @csenn -- yes, your side-project does sound interesting! Will it be available via an open source license?  If so, cool, I'd love to see it.  If it won't be available via open source, I need to determine if it's appropriate for us to engage at that level (just because of our position in this space and engagement w/ other companies).

As for the results, it might be hard to type those since CQL allows you to return arbitrary Tuples.  I suppose we could type the FHIR result types and authors can cast to them if they're sure of what it is.  The reason for the variation from FHIR is that the result types align with the CQL FHIR model info (which expresses FHIR primitives as complex types with a `value`) rather than the FHIR JSON serialization.  I've thought in the past that it might be nice to add a `toFHIRJSON` function on the types to allow for conversion.

Reply:
Ah interesting, that makes sense that it's the model info. A `toFHIRJSON` could likely be useful though... Maybe we can discuss how that could be best achieved and I can help look into it. 

I'm actually not 100% sure where the project I'm working on is going, more just trying to figure what makes sense right now. It will certainly be free, but probably a combination of open and closed source since there will need to be login, data persistence, CI pipeline, etc. If you do have a chance to take a quick gander once it's ready, you'll probably pretty immediately see why the output type matters. Since data IN is FHIR, it's a little hard to intuit that OUT comes a different data structure especially for common scenarios such as filtering Conditions by a ValueSet. But yeah like you mentioned, when shaping and returning Tuples you're actually not working with FHIR anymore, so it's tricky...

Reply:
On re-review (in an effort to button this PR up), I think our guidance is to update this code to doc why we have to `@ts-ignore` it and to add the eslint rule as Matt suggested.  This is a potential point for improvement in the future, but let's just leave it at that for now.